chain-extensions/xvm/src/lib.rs

```diff
@@ -98,8 +98,10 @@ where
                 // Claim the default evm address if needed.
                 let mut actual_weight = Weight::zero();
                 if value > 0 {
-                    // `UA::to_h160` 1 DB read.
-                    actual_weight.saturating_accrue(T::DbWeight::get().reads(1));
+                    // `UA::to_h160`.
+                    actual_weight.saturating_accrue(
+                        <T as pallet_unified_accounts::Config>::WeightInfo::uam_to_h160(),
+                    );
```

Comment:
Before calling `UA::to_h160` you need to check if `weight_limit` has enough funds to cover the cost.
Otherwise user could consume more resources than what's being paid for.

Same for the weight before `claim_default_evm_address` extrinsic is called.

Reply:
I think we discussed this in XVM PR with @shaunxw  - https://github.com/AstarNetwork/Astar/pull/1056#discussion_r1357658888

This PR just replace the hardcoded weights to the benchmarked ones.

Reply:
I see, I must have missed that discussion or forgot about it.
Anyhow, I'm aware what this PRs changing but that doesn't mean I cannot notice something wrong.

IMO, this should be changed - the solution doesn't have to be perfect, but it's better to slightly overcharge the user than to undercharge him. It's just _one call_ now, but over time with such practice it can amount to more.

Reply:
Since it was introduced in my XVM PR, let me create an issue and follow up.